### ISSUE TITLE
Use cnn for regression。 why Score from  0.144361  to Infinity and NaN

### ISSUE THREAD

**Issue by marshalWS** [deeplearning4j/deeplearning4j#4797]
#### Issue Description
**All  feature and Label are 0-1.** Maybe some configuration have errors。
 I use my training-datas on cnn models. 
 cnn- net config is: 
   `private MultiLayerConfiguration config(int channels, int seed, int outputNum) {

        Map<Integer, Double> lrSchedule = new HashMap<>();
        lrSchedule.put(0, 0.06);
        lrSchedule.put(200, 0.05);
        lrSchedule.put(600, 0.028);

        MultiLayerConfiguration multiLayerConfig = new NeuralNetConfiguration.Builder()
                .seed(seed)
                .iterations(iterations)
                .regularization(true).l2(0.0005)
                .learningRate(.01)
                .learningRateDecayPolicy(LearningRatePolicy.Schedule)
                .learningRateSchedule(lrSchedule)
                .weightInit(WeightInit.XAVIER)
                .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
                .updater(Updater.NESTEROVS)
                .list()
                .layer(0, new ConvolutionLayer.Builder(5, 5)
                        .nIn(channels)
                        .stride(1, 1)
                        .nOut(20)
                        .activation(Activation.RELU)
                        .build())
                .layer(1, new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX)
                        .kernelSize(2, 2)
                        .stride(2, 2)
                        .build())
                .layer(2, new ConvolutionLayer.Builder(5, 5)
                        .stride(1, 1)
                        .nOut(50)
                        .activation(Activation.RELU)
                        .build())
                .layer(3, new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX)
                        .kernelSize(2, 2)
                        .stride(2, 2)
                        .build())
                .layer(4, new DenseLayer.Builder().activation(Activation.RELU)
                        .nOut(500).build())
                .layer(5, new OutputLayer.Builder(LossFunctions.LossFunction.MSE)
                        .nOut(outputNum)
                        .activation(Activation.IDENTITY)
                        .build())

                .setInputType(InputType.convolutionalFlat(72, 55, 3))
                .backprop(true).pretrain(false).build();

        return multiLayerConfig;
    }`
`        CNNDataRecordReader trainRR = new CNNDataRecordReader(channels, height, width, normalizedCNNSamples.iterator());
        DataSetIterator trainIter = new CNNRecordReaderDataSetIterator(trainRR, batchSize, 1, 1, true);`

Please describe our issue, along with:
- expected behavior
        
- encountered behavior
        I  not use  image data , not for classify, just for regerssion, why console print class 0 & 1 ?  

        2018.03.13 16:33:04 org.nd4j.linalg.factory.Nd4jBackend - Loaded [CpuBackend] backend
2018.03.13 16:33:04 org.reflections.Reflections - Reflections took 388 ms to scan 149 urls, producing 56255 keys and 60572 values 
2018.03.13 16:33:05 org.nd4j.nativeblas.NativeOpsHolder - Number of threads used for NativeOps: 4
2018.03.13 16:33:05 org.reflections.Reflections - Reflections took 236 ms to scan 13 urls, producing 31 keys and 227 values 
2018.03.13 16:33:05 org.nd4j.nativeblas.Nd4jBlas - Number of threads used for BLAS: 4
2018.03.13 16:33:05 org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Backend used: [CPU]; OS: [Windows 7]
2018.03.13 16:33:05 org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Cores: [4]; Memory: [1.8GB];
2018.03.13 16:33:05 org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Blas vendor: [OPENBLAS]
2018.03.13 16:33:07 org.reflections.Reflections - Reflections took 1256 ms to scan 126 urls, producing 2480 keys and 10258 values 
2018.03.13 16:33:07 org.deeplearning4j.nn.multilayer.MultiLayerNetwork - Starting MultiLayerNetwork with WorkspaceModes set to [training: NONE; inference: SEPARATE]
2018.03.13 16:33:07 org.reflections.Reflections - Reflections took 156 ms to scan 13 urls, producing 387 keys and 1538 values 
2018.03.13 16:33:08 org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 0 is 0.14436129544540774
2018.03.13 16:33:08 org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1 is 11424.159089943982
2018.03.13 16:33:08 org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 2 is 1.242292317973531E16
2018.03.13 16:33:09 org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 3 is Infinity
2018.03.13 16:33:09 org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 4 is NaN
2018.03.13 16:33:10 com.feeyo.learn.sample.cnn.CNNModel - 
Examples labeled as 0 classified by model as 0: 49 times
Examples labeled as 1 classified by model as 0: 50 times

Warning: 1 class was never predicted by the model and was excluded from average precision
Classes excluded from average precision: [1]

==========================Scores========================================
 # of classes:    2
 Accuracy:        0.4949
 Precision:       0.4949
 Recall:          0.5000
 F1 Score:        0.0000
========================================================================

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version             0.9.1
* platform information (OS, etc)    win7
* CUDA version, if used
* NVIDIA driver version, if in use

#### Contributing

If you'd like to help us fix the issue by contributing some code, but would
like guidance or help in doing so, please mention it!
